### PR TITLE
Refactored the java event-list fragment and event-info activity

### PR DESCRIPTION
**Pull request by gdumouli**
Mostly there for review: the PR doesn't need to be urgently merged

### CODE REVIEW

File: code/noRAM/app/src/main/java/com/example/noram/EventListFragmentTemplate.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+ * Should be used to create fragments that contain lists of events
+ * @maintainer Gabriel
+ * @author Gabriel
+ */
+public abstract class EventListFragmentTemplate extends Fragment {
+    private ArrayList<Event> searchEventDataList; // data list of events' search results
+    EventArrayAdapter searchEventAdapter; // adapter for searchEvent list
+    private final CollectionReference eventRef = MainActivity.db.getEventsRef(); // list of events in database
+
+    protected void setupSearch(ListView searchList, EditText searchInput){
```
Comment (WhoShotCupidd) [inline code/noRAM/app/src/main/java/com/example/noram/EventListFragmentTemplate.java:43]: comment

File: code/noRAM/app/src/main/java/com/example/noram/EventListFragmentTemplate.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        // connect each element of the list to a function
+        searchList.setOnItemClickListener((parent, view, position, id) -> {
+            Event event = searchEventDataList.get(position);
+            searchElementsClick(event);
+        });
+
+        // connect searchbar to listen for user input
+        searchInput.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
```
Comment (WhoShotCupidd) [inline code/noRAM/app/src/main/java/com/example/noram/EventListFragmentTemplate.java:65]: comment all these plz

**Comment by WhoShotCupidd** [comment]
I just checked for formatting and comments. Besides the small comments I made above, it all looks good on that front. I didn't test functionality tho.

File: code/noRAM/app/src/main/java/com/example/noram/EventListFragmentTemplate.java (hunk 2 above)
Comment (gdumouli) [reply code/noRAM/app/src/main/java/com/example/noram/EventListFragmentTemplate.java:65]: Don't we use regular comments inside methods (including comments for listeners)?

Comment (WhoShotCupidd) [reply code/noRAM/app/src/main/java/com/example/noram/EventListFragmentTemplate.java:65]: whenever you are defining a function such as beforeTextChanged it needs a javadoc

**Review by ethanchung936 — APPROVED** [review]
LGTM

**Comment by ColeDewis** [comment]
@gdumouli this looks like it could potentially be useful for the Organizer page - do you think we will use it there as well? I've got a PR that's mostly ready for that stuff, but maybe we want to add this stuff in?

**Comment by gdumouli** [comment]
> @gdumouli this looks like it could potentially be useful for the Organizer page - do you think we will use it there as well? I've got a PR that's mostly ready for that stuff, but maybe we want to add this stuff in?

Yes, I think it could be used to list events in both the attendee and organizer sections. 
I can merge with master if you already merged your PR then update the branch

**Pull request merged**